lib/service/service.go

```diff
@@ -1156,8 +1156,9 @@ func (process *TeleportProcess) initAuthService() error {
 					Name:      process.Config.HostUUID,
 				},
 				Spec: services.ServerSpecV2{
-					Addr:     authAddr,
-					Hostname: process.Config.Hostname,
+					Addr:            authAddr,
+					Hostname:        process.Config.Hostname,
+					TeleportVersion: teleport.Version,
```

Comment:
I think `Teleport` is redundant here, we can assume that it's all teleport related

Reply:
I also think it's beneficial to report every component version with the heartbeat, so can you extend this to all other parts in the same file, proxy and nodes in particular.

Reply:
It implements resource interface where `version` has been reserved.

Reply:
But perhaps as part of this struct `type ServerV1 struct`, I agree it can be just the `Version`.

Reply:
@alex-kovoy I don't think `ServerSpecV2` uses `Version`.

Reply:
@russjones I was referring to this existing code

```golang
// GetVersion returns resource version
func (s *ServerV2) GetVersion() string {
	return s.Version
}
```

So maybe TeleportVersion is a good name after all.

Reply:
I will change the proto field to `Version` under `ServerSpecV2`, and keep the `GetTeleportVersion`. Version has been very confusing as it can mean both `resource version` or `teleport version`.